toolkit/tools/imagecustomizerapi/dirconfig.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+)
+
+// DirConfigList is a list of destination files where the source file will be copied to in the final image.
+// This type exists to allow a custom marshaller to be attached to it.
+type DirConfigList []DirConfig
+
+type DirConfig struct {
+	// The path in the target OS that the directory will be copied to.
+	SourcePath string `yaml:"SourcePath"`
+
+	// The path in the target OS that the directory will be copied to.
+	DestinationPath string `yaml:"DestinationPath"`
+
+	// The permissions to set on the directory.
+	Permissions *DirPermissions `yaml:"Permissions"`
```

Comment:
Looking at the implementation, `Permissions` seems to be the permissions that is given to top-level directory if it doesn't already exist. It might be good to clarify the comment.

Reply:
Added more clarity to the comment.